src/lib/net.js

```diff
@@ -15,731 +15,19 @@
-              if ( this.readyState === this.DONE )
-                resolve(this.response);
-            });
-          });
-      }
-    },
-    {
-      class: 'Int',
-      name: 'pos',
-      value: 0
-    }
-  ],
-
-  methods: [
-    function start() {
-      this.streaming = true;
-      this.xhr.addEventListener('loadend', function() {
-        this.done.pub();
-      }.bind(this));
-
-      this.xhr.addEventListener('progress', function() {
-        var substr = this.xhr.responseText.substring(this.pos);
-        this.pos = this.xhr.responseText.length;
-        this.data.pub(substr);
-
-        if ( this.pos > this.STREAMING_LIMIT ) {
-          this.xhr.abort();
-        }
-      }.bind(this));
-    }
-  ]
-});
-
-
-foam.CLASS({
-  package: 'foam.net',
-  name: 'SafariEventSource',
-  extends: 'foam.net.EventSource',
-
-  requires: [
-    'foam.net.XMLHTTPRequest as HTTPRequest'
-  ],
-
-  properties: [
-    {
-      class: 'String',
-      name: 'buffer'
-    }
-  ],
-
-  listeners: [
-    function onData(s, _, data) {
-      this.delay = 1;
-      this.keepAlive();
-
-      this.buffer += data;
-      var string = this.buffer;
-
-      while ( string.indexOf('\n') != -1 ) {
-        var line = string.substring(0, string.indexOf('\n'));
-        this.processLine(line);
-        string = string.substring(string.indexOf('\n') + 1);
-      }
-
-      this.buffer = string;
-    }
-  ]
-});
-
-
-foam.ENUM({
-  package: 'foam.net',
-  name: 'Environment',
-
-  values: [
-    { name: 'BROWSER', label: 'Browser' },
-    { name: 'NODE',    label: 'Node'    }
-  ]
-});
-
-
-foam.CLASS({
-  package: 'foam.net',
-  name: 'Universal',
-
-  axioms: [ foam.pattern.Singleton.create() ],
-  requires: [ 'foam.net.Environment' ],
-
-  properties: [
-    {
-      class: 'Enum',
-      of: 'foam.net.Environment',
-      name: 'environment',
-      factory: function() {
-        return typeof process === 'undefined' ?
-            this.Environment.BROWSER :
-            this.Environment.NODE;
-      }
-    }
-  ],
-
-  methods: [
-    function getRuntimeClass(cls) {
-        return foam.lookup((this.environment === this.Environment.NODE ?
-            'foam.net.node' : 'foam.net') + '.' + cls.name);
-    },
-    function installInClass(cls) {
-      var runtimeCls = this.getRuntimeClass(cls);
-      cls.create = function(args, opt_parent) {
-        return runtimeCls.create(args, opt_parent);
-      };
-    }
-  ]
-});
-
-
-foam.CLASS({
-  package: 'foam.net.universal',
-  name: 'HTTPRequest',
-
-  axioms: [ foam.net.Universal.create() ]
-});
-
-
-foam.CLASS({
-  package: 'foam.net.universal',
-  name: 'HTTPResponse',
-
-  axioms: [ foam.net.Universal.create() ]
-});
+(function() {
+  var pkg = 'foam.net.' + (typeof process === 'undefined' ? 'web' : 'node');
+  console.log('PKG:', pkg);
+  var clss = [
+    'HTTPRequest',
+    'HTTPResponse'
+  ];
+
+  // For each class with a "web" (browser) and "node" (Node JS)
```

Comment:
Add a TODO this should be provided via some sort of ContextFactory IMO.

Also your comment is not entirely accurate, since you are not doing this for the WebSocketService class, which might be good to do.